### ISSUE TITLE
[native-image] the agent includes $ chars in the middle of patterns in generated resource-config.json

### ISSUE THREAD

**Issue by aclement** [oracle/graal#1949]
Spring (currently) loads .class files as resources (for processing with asm). This means when you use the agent you get entries in the resource-config.json like this if you are loading nested classes (notice the '$' in there):

{"pattern":"org/springframework/boot/autoconfigure/cache/CacheAutoConfiguration$CacheConfigurationImportSelector.class"},   {"pattern":"org/springframework/boot/autoconfigure/cache/CacheAutoConfiguration$CacheManagerEntityManagerFactoryDependsOnPostProcessor.class"},    {"pattern":"org/springframework/boot/autoconfigure/cache/CacheAutoConfiguration$CacheManagerValidator.class"},

Those references don't work and you get errors at runtime about missing resources (presumably because $ is being considered 'end of input' pattern marker).

```
Caused by: java.io.FileNotFoundException: class path resource [org/springframework/boot/autoconfigure/web/reactive/function/client/ClientHttpConnectorConfiguration$ReactorNetty.class] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180) ~[na:na]
```

A crude workaround is to manually modify the $ to '.' (presumably '.' then meaning any character as this is a pattern, so it does match the $ in the .class resource name). The right answer is no doubt some form of escaping.

**Comment by peter-hofer** [comment]
Thank you for your report @aclement ! Indeed the agent needs to escape these patterns. I will look into it.

**Comment by peter-hofer** [comment]
Fixed in eaa723ac42f9ae902d9801ad2cb19c194c62c283.
